src/bitcoin/wallet.rs

```diff
@@ -254,16 +259,22 @@ impl Wallet {
             hash[0], hash[1], hash[2], hash[3]
         )
     }
+
+    pub fn seed(&self) -> Seed {
```

Comment:
I see this is done to generate a secret hash.
It is strange because the seed is being taken from the Bitcoin Wallet, I now realize that the seed stays in memory which is not great for security concern.

If you need a `Seed` to generate the secret then pass it direcly where you need it instead of having this indirection `Seed -> BitcoinWallet -> Seed -> SecretHash`.

Maybe better to actually hash the seed and use the hash of the seed as a seed for the secret hash. The seed is the most important secret here so we should avoid passing it around.

Reply:
Agree with the hash idea. Is it ok if this is done in a seperate PR?
The indirection is there because I did not want to duplicate the seed in the Nectar struct. I had initially done it this way

Reply:
Yes, feel free to clean up in a separate PR. Adding a `TODO` here would be a good way to track a low hanging fruit clean up.

Reply:
there is a todo in the Nectar struct which will link to this.